ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -813,34 +813,24 @@ encodeByronGenTxId genTxId = case genTxId of
 encodeByronApplyTxError :: ApplyTxErr (ByronBlockOrEBB cfg) -> Encoding
 encodeByronApplyTxError = toCBOR
 
+-- | The 'ByteString' annotation will be the canonical encoding.
```

Comment:
While thinking about #905 with @edsko, we realised some things. Replace the docstring with the following:
```haskell
-- | The 'ByteString' annotation will be the canonical encoding.
--
-- While the new implementation does not care about canonical encodings, the
-- old one does. When a generalised transaction arrives that is not in its
-- canonical encoding (only the 'CC.UTxO.ATxAux' of the 'ByronTx' can be
-- produced by nodes that are not under our control), the old implementation
-- will reject it. Therefore, we need to reject them too. See #905.
--
-- We use the ledger to check for canonical encodings: the ledger will check
-- whether the signed hash of the transaction (in the case of a
-- 'CC.UTxO.ATxAux', the transaction witness) matches the annotated
-- bytestring. Is therefore __important__ that the annotated bytestring be the
-- /canonical/ encoding, not the /original, possibly non-canonical/ encoding.
```